src/HighLevel/StatementBuilder.js

```diff
@@ -179,33 +177,47 @@ export class StatementBuilder {
     }
     return this;
   }
+  
+  /**
+   * Sends a statement to the queue and returns a promise that resolves when the statement is processed.
+   *
+   * @returns {Promise<void>} The promise sent
+   */
+  Send() {
+    if (!this._sendPromise) {
+      this._sendPromise = this.client.enqueue(this.statement);
+    }
+    return this._sendPromise;
+  }
 
-  //— make this builder awaitable (thenable) ————————————————————————
   /**
-   * 
-   * @param {*} onFulfilled
-   * @param {*} onRejected 
-   * @returns {Promise<void>}
+   * Makes this builder awaitable (thenable).
+   *
+   * @param {(value: void) =>void | PromiseLike<void>} onFulfilled - Called when the promise is resolved
+   * @param {(reason:any)=>void | PromiseLike<void>} onRejected - Called when the promise is rejected
+   * @returns {Promise<void>} The resulting promise
    */
   then(onFulfilled, onRejected) {
```

Comment:
Not using builder pattern.... Using then / catch / finally is strange outside the context of Promises. This must be refactored

Reply:
removed